kv/memberlist/tcp_transport_test.go

```diff
@@ -61,6 +67,58 @@ func TestTCPTransport_WriteTo_ShouldNotLogAsWarningExpectedFailures(t *testing.T
 	}
 }
 
+type timeoutReader struct{}
+
+func (f *timeoutReader) ReadSecret(_ string) ([]byte, error) {
+	time.Sleep(1 * time.Second)
+	return nil, nil
+}
+
+func TestTCPTransportWriteToUnreachableAddr(t *testing.T) {
+	writeCt := 50
+
+	// Listen for TCP connections on a random port
+	freePorts, err := getFreePorts(1)
+	require.NoError(t, err)
+	addr := &net.TCPAddr{IP: net.IPv4(127, 0, 0, 1), Port: freePorts[0]}
+	listener, err := net.ListenTCP("tcp", addr)
```

Comment:
```suggestion
	// Listen for TCP connections on a random port
	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

and then later `listener.Addr().String()` instead of `addr.String()`.